pkg/docker/docker_test.go

```diff
@@ -24,201 +30,336 @@ func TestContainerName(t *testing.T) {
 }
 
 func getDocker(client Client) *stiDocker {
+	//k8s has its own fake docker client mechanism
+	k8sDocker := dockertools.ConnectToDockerOrDie("fake://")
 	return &stiDocker{
-		client:   client,
-		pullAuth: docker.AuthConfiguration{},
+		kubeDockerClient: k8sDocker,
+		client:           client,
+		pullAuth:         dockertypes.AuthConfig{},
 	}
 }
 
+//NOTE, neither k8s kube client nor engine api make their error types public, so we can access / instantiate them like we did for the analogous
+// ones in go-dockerclient.
+// Hence, we test with errors created in these tests and set on the fake client  vs. those actually defined in k8s/engine-api.
+// Also, the fake client does save the parameters passed in; but instead, you can confirm that it was called via AssertCalls
```

Comment:
and the fake docker client is under our control, so why don't we update it to save the parameters?